Feature/join us

Vish-p0 opened the pull request:
Join Us Form – Google Sheets Integration

Added Join Us form page in app/join-us/page.tsx with fields:
timestamp, email, fullName, studentId, academicYear, semester, specialization, whatsapp, linkedin, github, reason, otherClubs, preferredTeam, applicationStatus

Created API route (/api/join-us) to handle submissions and write to Google Sheets

Configured Google Sheets API via service account credentials + env variables

Submissions are now saved to Google Sheets in real-time

Updated Navbar: rerouted Join Us link from external Google Form to the new page

Added form validation, success screen, and error handling

Installed googleapis dependency

Removed static export (output: "export") to enable API routes

Vish-p0 committed:
Join us form

Vish-p0 committed:
Join us form validations

Seniru commented on app/join-us/page.tsx:
```diff
@@ -0,0 +1,717 @@
+              aria-describedby={errors.whatsappNumber && touched.whatsappNumber ? "whatsappNumber-error" : undefined}
+            />
+          </div>
+          <datalist id="country-codes">
```
Oh I'm pretty sure everyone is from Sri Lanka 😛

Vish-p0 replied:
just felt like adding it cause whatsapp numbers have the country code soo..... ill remove it if its not needed

Seniru replied:
Nah it's fine. I didn't check the form yet. We'll reconsider if selecting it felt like much of a hassle.
Just curious, are there people using anything other than 94 code here, maybe some business people I guess?

Vish-p0 replied:
im not sure if any o them are in the group but people tend to use business accounts soo thought might as well

Seniru commented:
Nice stuff @Vish-p0! Can you specify the configuration steps for this feature? It's easier when things are documented

Vish-p0 commented:
This document explains how to prepare your environment and Google services before testing the **Join Us form with Google Sheets integration**.

## 1. Clone the Project

```bash
git clone https://github.com/Mozilla-Campus-Club-of-SLIIT/official-web.git
cd official-web
```

## 2. Install Dependencies

```bash
npm install
npm install googleapis
```

(We use the `googleapis` library to connect to Google Sheets.)

## 3. Environment Variables

Create a `.env.local` file in the project root with the following variables:
```python
GOOGLE_CLIENT_EMAIL=your-service-account@project-id.iam.gserviceaccount.com
GOOGLE_PRIVATE_KEY="-----BEGIN PRIVATE KEY-----\nABC123...\n-----END PRIVATE KEY-----\n"
GOOGLE_SHEET_ID=your-google-sheet-id
```

- **GOOGLE_CLIENT_EMAIL** → Found in your Service Account JSON (`client_email`).
- **GOOGLE_PRIVATE_KEY** → Found in your Service Account JSON (`private_key`). Replace line breaks with `\n`.
- **GOOGLE_SHEET_ID** → The ID part of your Google Sheet URL.


## 2. Get Access to google sheet via invite link

## 3 . Run the Dev Server

```bash
npm run dev
```

- Open http://localhost:3000/join-us

## 4. Submit a Test Form

- Fill in the fields (Name, Email, Student ID, etc.).
- Click **Submit**.
- You should see a **success message** if submission worked.

---

## 5. Verify in Google Sheets

- Check the Google Sheet (link will be provided by the maintainer).
- Confirm your test submission appears as a new row.

## 6. Troubleshooting

- If you see `PERMISSION_DENIED` → the service account might not have access. Contact the maintainer.
- If you see `Unexpected token '<'` → make sure you are not using `output: "export"` in `next.config.js`.
- If nothing appears → double-check your `.env.local` values and restart the dev server.

Seniru reviewed (changes requested):
Some issue, when I click on one team (only 1 team) in preferred team section, it gives me an error - "select atleast one team". Ofcourse I can submit it with no issue. However it might create some confusion to some people. If you can, please fix it.

You can leave year, semester, and specialization and submit the form. It catches the error! However, it doesn't really show where. It just says "Please fix the errors", not where. Please take a look into that too.

Not really a big fan of the background color of the success card. Try to use the colors we have used everywhere else. You can check cards in blog page. They are "whiteish".

Also a little request if possible. We can improve the validations llike this.
- Student ID validation: It's going to be a several combination of text. Use some regex like this to validate: `(IT|EN|BS|HM)\d{8}`
- Github, Linkedin links: I can submit ANY link. You can use some form of regex or `startswith` method here to make sure they are only github or linkedin links

The above validations are optional. We can leave it if you don't want to implement them right now.



Rest is awesome! Nice work.

Vish-p0 committed:
fix validation issues and success message

Vish-p0 commented:
# Update: Addressed feedback

1. **Success Message Card**
   - Replaced the plain background with a `<Card>` styled like the blog cards.
   - Card includes a title, thank-you copy, WhatsApp group link, and a “Return home” button.
   - Reduced vertical spacing (`py-8` instead of `min-h-screen`) for a compact, consistent look.

2. **Student ID Input Restriction**
   - Field limited to **max 10 chars**.
   - Enforces prefix from **IT/EN/BS/HS** followed by **8 digits**.
   - Input auto-uppercases and filters invalid characters as the user types.

3. **LinkedIn & GitHub URL Validation**
   - Accepts URLs **with or without** `http://` / `https://`.
   - Regex/domain checks allow:
     - `linkedin.com/in/username`, `www.linkedin.com/in/...`, and full LinkedIn URLs
     - `github.com/username` and full GitHub URLs
   - Shows inline errors on invalid URLs; aligned with backend validation.

4. **Preferred Teams Validation**
   - Requires **at least one** preferred team.
   - Error clears immediately once a valid selection is made.

5. **Field-Level & Global Error Handling**
   - Required/invalid fields now show **specific inline messages** next to each field.
   - Global footer message now reads: `Please fill all required fields marked with *`.
   - Backend field errors (e.g., LinkedIn/GitHub) are mapped to the correct inputs and shown inline.

6. **General UI/UX**
   - All `*` fields are required; validation runs on **blur** and on **submit**.

Seniru reviewed (dismissed):
LGTM

Seniru committed:
Merge branch 'main' into feature/joinUs

Seniru reviewed (changes requested):
Looks good now! Just one thing, could you format the code please. Use `npm run format`.

I also did something with the package lock file to resolve the merge conflicts. Can you check if it's still valid. If it shows no errors then it's working fine

Vish-p0 committed:
style(join-form): format files with Prettier

Vish-p0 commented:
**Update:** Regenerated and validated `package-lock.json` locally (`npm ci` OK).  
Ran the formatter and fixed Prettier issues.  
Ready for re-review — thanks!

Seniru reviewed (approved):
Awesome, thank you for working on this PR! We are using conventional commit guidelines for commit messages so make sure your messages align with them in the future!
